code/datums/components/README.md

```diff
@@ -0,0 +1,102 @@
+        * `COMPONENT_DUPE_UNIQUE`: New component will be deleted, old component will first have `/datum/component/proc/InheritComponent(datum/component/new, TRUE)` on it
+1. `/datum/component/var/list/signal_procs` (private)
+    * Associated lazy list of signals -> callbacks that will be run when the parent datum recieves that signal
+1. `/datum/component/var/datum/parent` (protected, read-only)
+    * The datum this component belongs to
+
+### Procs
+
+1. `/datum/proc/GetComponent(component_type(type)) -> datum/component?` (public, final)
+    * Returns a reference to a component of component_type if it exists in the datum, null otherwise
+1. `/datum/proc/GetComponents(component_type(type)) -> list` (public, final)
+    * Returns a list of references to all components of component_type that exist in the datum
+1. `/datum/proc/GetExactComponent(component_type(type)) -> datum/component?` (public, final)
+    * Returns a reference to a component whose type MATCHES component_type if that component exists in the datum, null otherwise
+1. `GET_COMPONENT(varname, component_type)` OR `GET_COMPONENT_FROM(varname, component_type, src)`
+    * Shorthand for `var/component_type/varname = src.GetComponent(component_type)`
+1. `/datum/proc/AddComponent(component_type(type), ...) -> datum/component`  (public, final)
+    * Creates an instance of `component_type` in the datum and passes `...` to it's `New()` call
+    * Sends the `COMSIG_COMPONENT_ADDED` signal to the datum
+    * All components a datum owns are deleted with the datum
+    * Returns the component that was created. Or the old component in a dupe situation where `COMPONENT_DUPE_UNIQUE` was set
+1. `/datum/proc/RemoveComponent(datum/component/C)` (public, final)
+    * Sends the `COMSIG_COMPONENT_REMOVING` signal to the datum
+    * Deletes the passed component
+    * Component deleting must use this
+1. `/datum/proc/ComponentActivated(datum/component/C)` (abstract)
+    * Called on a component's `parent` after a signal recieved causes it to activate. `src` is the parameter
+    * Will only be called if a component's callback returns `TRUE`
+1. `/datum/proc/TakeComponent(datum/component/C)` (public, final)
+    * Properly transfers ownership of a component from one datum to another
+    * Singals `COMSIG_COMPONENT_REMOVING` on the parent
+    * Called on the datum you want to own the component with another datum's component
+1. `/datum/proc/SendSignal(signal, ...)` (public, final)
+    * Call to send a signal to the components of the target datum
+    * Extra arguments are to be specified in the signal definition
+1. `/datum/component/New(datum/parent, ...)` (protected, virtual)
+    * Forwarded the arguments from `AddComponent()`
+1. `/datum/component/Destroy()` (private)
+    * Use `RemoveComponent()` instead of `qdel()` to properly delete a component
+    * `qdel()` still works but it suppresses the `COMSIG_COMPONENT_REMOVING` signal and is not intended behavior

```

Comment:
Yeah, I'm trying to figure out how without adding another var, but it seems unlikely